src/modules/users/components/UserProfile/UserProfile.tsx

```diff
@@ -20,21 +19,23 @@ const UserProfile = ({
     params: { username },
   },
 }: Props) => {
-  const { userAddress, error: userAddressError } = useUserAddressFetcher(
```

Comment:
We're also using the `useUserAddressFetcher` custom hook in the core `UserMention` component.

It would be good if you refactored that one in the same style as you did here with `useUserAddressQuery`, and after that you can remove `useUserAddressFetcher` from the code entirely since it won't be used anymore.